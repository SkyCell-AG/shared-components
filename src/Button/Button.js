/* eslint-disable complexity */
import React from 'react'
import PropTypes from 'prop-types'
import clsx from 'clsx'
import {
    CircularProgress,
} from '@material-ui/core'
import noop from 'lodash/noop'

import useStyles from './Button.style'

const propTypes = {
    className: PropTypes.string,
    label: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired,
    secondary: PropTypes.bool,
    disabled: PropTypes.bool,
    saving: PropTypes.bool,
    icon: PropTypes.element,
    isUpperCase: PropTypes.bool,
}

const defaultProps = {
    className: '',
    secondary: false,
    disabled: false,
    saving: false,
    icon: null,
    isUpperCase: true,
}

const Button = ({
    label,
    onClick,
    secondary,
    disabled,
    saving,
    icon,
    className,
    isUpperCase,
}) => {
    const classes = useStyles()

    return (
        <div className={clsx(classes.buttonContainer, className)}>
            <div
                data-testid={`button-${label}`}
                className={clsx(!secondary && classes.buttonPrimary,
                    secondary && classes.buttonSecondary,
                    classes.button,
                    disabled && classes.disabled)}
                onMouseDown={disabled ? noop : onClick}
            >
                {saving && (
                    <>
                        <CircularProgress
                            size={16}
                            color="inherit"
                        />
                        &nbsp;
                    </>
                )}
                {icon}
                &nbsp;
                <div className={clsx(
                    isUpperCase && classes.buttonTitleUppercase,
                    classes.buttonTitle,
                )}
                >
                    {label}
                </div>
            </div>
        </div>

    )
}

Button.propTypes = propTypes
Button.defaultProps = defaultProps

export default Button
