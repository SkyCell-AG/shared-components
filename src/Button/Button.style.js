import {
    makeStyles,
} from '@mui/styles'
import {
    useTheme,
} from '@mui/material/styles'

const useStyles = makeStyles(() => {
    const theme = useTheme()

    return {
        buttonContainer: {
            display: 'flex',
            flexDirection: 'column',
            minWidth: 120,
        },
        button: {
            cursor: 'pointer',
            display: 'flex',
            justifyContent: 'center',
            alignItems: 'center',
            fontSize: theme.typography.body1.fontSize,
            fontFamily: theme.typography.fontFamily,
            height: 36,
            fontWeight: theme.typography.body1.fontWeight,
            letterSpacing: 1.25,
            borderRadius: 4,
            padding: `0px ${theme.spacing(0.5)}px`,
        },
        buttonPrimary: {
            background: theme.palette.primary[200],
            boxShadow: `0px 1px 3px ${theme.palette.secondary[400]}`,
            color: theme.palette.common.white,
        },
        buttonSecondary: {
            background: theme.palette.common.white,
            border: `1px solid ${theme.palette.primary[200]}`,
            color: theme.palette.primary[200],
        },
        disabled: {
            opacity: 0.5,
        },
        buttonTitle: {
            fontFamily: theme.typography.fontFamily,
            fontSize: theme.typography.button.fontSize,
            fontWeight: theme.typography.button.fontWeight,
            letterSpacing: theme.typography.button.letterSpacing,
        },
        buttonTitleUppercase: {
            textTransform: 'uppercase',
        },
    }
})

export default useStyles
