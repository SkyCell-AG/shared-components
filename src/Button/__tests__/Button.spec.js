import React from 'react'
import {
    shallow,
} from 'enzyme'

import Button from '../Button'

it('Button snapshot', () => {
    const wrapper = shallow(
        <Button />,
    )

    expect(wrapper).toMatchSnapshot()
})
