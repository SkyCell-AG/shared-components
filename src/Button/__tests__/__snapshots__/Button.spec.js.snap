// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot 1`] = `
<div
  className="makeStyles-buttonContainer-1"
>
  <div
    className="makeStyles-buttonPrimary-3 makeStyles-button-2"
    data-testid="button-undefined"
  >
    Â 
    <div
      className="makeStyles-buttonTitleUppercase-7 makeStyles-buttonTitle-6"
    />
  </div>
</div>
`;
